Sources/TemplateKit/Data/TemplateDataEncoder.swift

```diff
@@ -4,13 +4,13 @@ public final class TemplateDataEncoder {
     public init() {}
 
     /// Encode an `Encodable` item to `TemplateData`.
-    public func encode<E>(_ encodable: E, on worker: Worker) throws -> Future<TemplateData> where E: Encodable {
+    public func encode<E>(_ encodable: E, on worker: Worker, userInfo: [CodingUserInfoKey: Any] = [:]) throws -> Future<TemplateData> where E: Encodable {
```

Comment:
I too thought about mapping the entire dictionary to a single key.  I chose against it because it makes the semantics of the Encoder.userInfo call a bit odd:

```swift
    let myUserInfo: [AnyHashable: Any] = [
        "myKey" : "myData"
    ]
    
    renderer.render(path, context, userInfo: myUserInfo)
    
    public func encode(to encoder: Encoder) throws {
        let myData = (encoder.userInfo[CodingUserInfoKey(rawValue: "renderUserInfo")] as? [AnyHashable: Any])["myKey"]
```

However, whatever you think is best.  Just let me know and I'll align with your suggestion.

As for the init(), no problem I'll change it.